File: app/src/main/java/com/pavelprymak/propodcast/presentation/adapters/PodcastInfoClickListener.java
package com.pavelprymak.propodcast.presentation.adapters;

import android.view.View;

import com.pavelprymak.propodcast.model.network.pojo.podcastById.EpisodesItem;
import com.pavelprymak.propodcast.model.network.pojo.podcasts.PodcastItem;

public interface PodcastInfoClickListener {
    void onEpisodeItemClick(EpisodesItem episodesItem);

    void onMoreEpisodeClick();

    void onEpisodeMoreOptionClick(EpisodesItem episodesItem, View view);

    void onRecommendationItemClick(String podcastId);

    void onPodcastMoreOptionsClick(PodcastItem podcastItem, View v);
}

File: app/src/main/java/com/pavelprymak/propodcast/presentation/adapters/PodcastInfoClickListener.kt
package com.pavelprymak.propodcast.presentation.adapters

import android.view.View

import com.pavelprymak.propodcast.model.network.pojo.podcastById.EpisodesItem
import com.pavelprymak.propodcast.model.network.pojo.podcasts.PodcastItem

interface PodcastInfoClickListener {
    fun onEpisodeItemClick(episodesItem: EpisodesItem)

    fun onMoreEpisodeClick()

    fun onEpisodeMoreOptionClick(episodesItem: EpisodesItem, view: View)

    fun onRecommendationItemClick(podcastId: String)

    fun onPodcastMoreOptionsClick(podcastItem: PodcastItem, v: View)
}
